### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by gaborcsardi** [r-lib/pak#146]
```r
❯ pak::pkg_install("r-lib/revdepcheck")
✔ Downloaded metadata files, 3.5 MB in 7 files.
Error: callr subprocess failed: $ operator is invalid for atomic vectors
```

```r
❯ err <- tryCatch(pak::pkg_install("r-lib/revdepcheck"), error = identity)
❯ err$parent$trace

 ERROR TRACE for simpleError

 12. (function (...)  ...
 13. base:::withCallingHandlers(cliapp_message = function(msg) { ...
 14. get("pkg_install_make_plan", asNamespace("pak"))(...)
 15. r$solve(policy = policy)
    R/package.R:212:3
 16. pkgdepends:::remotes_solve(self, private, match.arg(policy))
    R/remotes.R:123:15
 17. private$create_lp_problem(pkgs, policy)
    R/solve.R:15:3
 18. pkgdepends:::remotes__create_lp_problem(self, private, pkgs,  ...
    R/remotes.R:168:27
 19. pkgdepends:::remotes_i_create_lp_problem(pkgs, policy)
    R/solve.R:89:3
 20. pkgdepends:::remotes_i_lp_dependencies(lp)
    R/solve.R:126:3
 21. base:::lapply(setdiff(seq_len(num_candidates), ruled_out), depconds)
    R/solve.R:301:3
 22. pkgdepends:::FUN(X[[i]], ...)
 23. base:::Filter(x = cand, function(c) { ...
    R/solve.R:268:7
 24. base:::unlist(lapply(x, f))
 25. base:::lapply(x, f)
 26. pkgdepends:::FUN(X[[i]], ...)
 27. base:::isTRUE(satisfies_remote(res, pkgs[c, ]))
    R/solve.R:272:11
 28. pkgdepends:::satisfies_remote(res, pkgs[c, ])
    R/solve.R:272:11
 29. pkgdepends:::sat(resolution, candidate, config, ...)
    R/satisfies.R:8:3
 30. base:::.handleSimpleError(function (e)  ...
    R/type-github.R:136:3
 31. h(simpleError(msg, call))
```

**Comment by blairj09** [comment]
Perhaps related:
```

**Comment by blairj09** [comment]
Perhaps related:
```r
> pak::pkg_install("rstudio/chromote", ask = FALSE)
ℹ Checking for package metadata updates
✔ All 12 metadata files are current.
✔ Using session cached package metadata
✔ Using cached package metadata

→ Will install 1 packages:
  rstudio/chromote
  
→ Will update 2 packages:
  r-lib/later, rstudio/promises
  
→ Will not update 10 packages.

→ Will download 3 packages with unknown size.

ℹ Building later 0.8.0.9002
✔ Built later 0.8.0.9002 [13.2s]
Error: callr subprocess failed: Cannot open lock file: Permission denied
```

```r
> err <- tryCatch(pak::pkg_install("rstudio/chromote", ask = FALSE), error = identity)

> err$parent$trace

 ERROR TRACE for simpleError

 12. (function (...)  ...
 13. base:::withCallingHandlers(cliapp_message = function(msg) { ...
 14. get("pkg_install_do_plan", asNamespace("pak"))(...)
 15. pak:::install_package_plan(plan = plan, lib = lib, num_workers = num_workers)
 16. base:::withCallingHandlers({ ...
 17. pak:::handle_events(state, events)
 18. pak:::handle_event(state, i)
 19. proc$get_result()
 20. processx:::process_get_result(self, private)
 21. private$post_process()
 22. pak:::install_extracted_binary(filename, lib_cache, pkg_cache,  ...
 23. pak:::lock_cache(lib_cache, pkg_name, getOption("install.lock"))
 24. filelock::lock(lockfile)
 25. base:::.handleSimpleError(function (e)  ...
 26. h(simpleError(msg, call))

 x Cannot open lock file: Permission denied 
```

<details><summary>Details</summary>
<p>

> MacOS 10.14.6
> RStudio 1.3.335-1

Session Info:
``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       macOS Mojave 10.14.6        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Denver              
#>  date     2019-07-31                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                          
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.5.3)                  
#>  backports     1.1.4      2019-04-10 [1] CRAN (R 3.5.3)                  
#>  callr         3.3.1      2019-07-18 [1] CRAN (R 3.6.0)                  
#>  cli           1.1.0      2019-03-19 [1] CRAN (R 3.5.3)                  
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.0)                  
#>  desc          1.2.0      2019-05-13 [1] Github (r-lib/desc@c860e7b)     
#>  devtools      2.1.0      2019-07-06 [1] CRAN (R 3.6.0)                  
#>  digest        0.6.20     2019-07-04 [1] CRAN (R 3.6.0)                  
#>  evaluate      0.14       2019-05-28 [1] CRAN (R 3.6.0)                  
#>  fs            1.3.1      2019-05-06 [1] CRAN (R 3.6.0)                  
#>  glue          1.3.1      2019-03-12 [1] CRAN (R 3.5.2)                  
#>  highr         0.8        2019-03-20 [1] CRAN (R 3.5.3)                  
#>  htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.0)                  
#>  knitr         1.23       2019-05-18 [1] CRAN (R 3.6.0)                  
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.0)                  
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.0)                  
#>  pkgbuild      1.0.3      2019-03-20 [1] CRAN (R 3.5.3)                  
#>  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.1)                  
#>  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.0)                  
#>  processx      3.4.1      2019-07-18 [1] CRAN (R 3.6.0)                  
#>  ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.0)                  
#>  R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.2)                  
#>  Rcpp          1.0.2      2019-07-25 [1] CRAN (R 3.6.0)                  
#>  remotes       2.1.0      2019-06-24 [1] CRAN (R 3.6.0)                  
#>  rlang         0.4.0      2019-06-25 [1] CRAN (R 3.6.0)                  
#>  rmarkdown     1.14       2019-07-12 [1] CRAN (R 3.6.0)                  
#>  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.0)                  
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.0)                  
#>  stringi       1.4.3      2019-03-12 [1] CRAN (R 3.5.2)                  
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.5.2)                  
#>  testthat      2.2.1      2019-07-25 [1] CRAN (R 3.6.0)                  
#>  usethis       1.5.1      2019-07-04 [1] CRAN (R 3.6.0)                  
#>  withr         2.1.2.9000 2019-05-30 [1] Github (jimhester/withr@79c5e36)
#>  xfun          0.8        2019-06-25 [1] CRAN (R 3.6.0)                  
#>  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.1)                  
#> 
#> [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```
</p>
</details>

**Comment by gaborcsardi** [comment]
Fixed at 770d588cf71c1941da08ae1c6bef53787794d7bb and 8bfd64cdad4d29b72073c4146da81e85831e2607
